### PR TITLE
feat(encounter/v2): TakeAction + EndTurn handlers + NPC dispatch loop + combat translator (#505)

### LINKED ISSUE #505: feat(encounter/v2): TakeAction + EndTurn handlers + NPC dispatch loop + combat translators
## Goal

Implement the v1alpha2 `TakeAction` and `EndTurn` RPCs. Today both are registered with `codes.Unimplemented`. After this issue: clients can call `TakeAction(actor, action_ref={attack}, target.entity_id=monster)` to resolve attacks via the toolkit, and `EndTurn(actor)` to cycle initiative. After EndTurn cycles to an NPC, the rpg-api handler dispatches `Encounter.NPCAct(npcID)` to run the monster's turn (toolkit-driven), then continues cycling until a player is active again. Both RPCs return `FailedPrecondition` when called outside `TURN_BASED` mode.

This is the orchestrator slice of Wave 2.8 (Combat slice) on rpg-project board #11. Pairs with the rpg-toolkit slice (`rpg-toolkit#633`) and the rpg-dnd5e-web consumer slice (filed separately).

## What to read first

**Required role context** — fresh agents picking this up should boot via the rpg-api-member role and read these context entries from `docs/teams/roles/rpg-api-member/context/`:
- `patterns.json`: `pat-v2-handler-direct-no-orchestrator`, `pat-v2-directory-layout`, `pat-v2-broker-process-scoped`, `pat-v2-translator-typed-errors`, `pat-v2-bufconn-test-pattern`, `pat-v2-status-code-mapping`, `pat-v2-cause-effect-event-split` (from Wave 2.7).
- `discoveries.json`: `disc-v2-projectFor-helper-opportunity` and any Wave 2.6/2.7 entries.
- `dependencies.json`: dep-007 (rpg-toolkit/encounter — bumps to v0.3.0 after rpg-toolkit#633 merges), dep-008 (proto pin).
- Wave plan: `rpg-project/ideas/encounter/v1alpha2/plans/06-wave-2.8-combat-slice.md`

**Slice references — already in role context:**
- `internal/handlers/dnd5e/v2/encounter/handler.go` — Handler struct + `MoveEntity` template.
- `internal/handlers/dnd5e/v2/encounter/translate.go` — translator file with typed errors (`ErrViewerSawNothing`, `ErrUnknownEventType`).
- `internal/handlers/dnd5e/v2/encounter/project.go` — projection helper used by stream.
- `internal/integration/encounter_v2_test.go` — `EncounterV2IntegrationSuite` to extend.

**Toolkit upstream (depends on rpg-toolkit#633 merging first):**
- `rpg-toolkit/encounter` `v0.3.0` (tagged after rpg-toolkit#633) — adds `Encounter.SetMode`, `Encounter.TakeAction`, `Encounter.EndTurn`, `Encounter.NPCAct`, plus `MonsterData`, `Mode`, initiative on `Data`, plus events `AttackResolvedEvent`, `DamageDealtEvent`, `ConditionAppliedEvent`, `ModeChangedEvent`, `TurnStartedEvent`, `TurnEndedEvent`.

## Proto request/response (already in v1alpha2)

From `gen/go/dnd5e/api/v1alpha2/encounter/service.proto:66-100, 136-138`:

```proto
message TakeActionRequest {
  string encounter_id = 1;
  string actor_entity_id = 2;
  Ref action_ref = 3;        // {module:"dnd5e", type:"action", id:"attack"}
  ActionTarget target = 4;
}
message TakeActionResponse {
  optional InputRequired input_required = 1;
}

message EndTurnRequest {
  string encounter_id = 1;
  string entity_id = 2;
}
message EndTurnResponse {}

// Both gated:
// rpc TakeAction(TakeActionRequest) returns (TakeActionResponse);
// rpc EndTurn(EndTurnRequest) returns (EndTurnResponse);
// (Comment: "Turn-based only — server returns FailedPrecondition if mode != TURN_BASED.")
```

`ActionTarget` oneof: `entity_id`, `position`, `area`, `self`. For Wave 2.8, only `entity_id` (monster id) is wired.

### 1. `internal/handlers/dnd5e/v2/encounter/take_action.go` (new file)

```go
func (h *Handler) TakeAction(ctx context.Context, req *encounterv2pb.TakeActionRequest) (*encounterv2pb.TakeActionResponse, error)
```

Required behavior:
- Auth: `auth.GetPlayerID(ctx)` → empty returns `Unauthenticated`.
- Validation: `EncounterId`/`ActorEntityId` empty → `InvalidArgument`. `ActionRef` nil → `InvalidArgument`. `Target` nil or `Target.GetKind() == nil` → `InvalidArgument`.
- Load encounter data via `h.encRepo.Get(ctx, req.GetEncounterId())`. Missing → `NotFound`.
- `enc, err := encounter.LoadFromData(data, h.broker)` → `Internal` on error.
- **Mode gate** — if `enc.Mode() != core.ModeTurnBased` → `codes.FailedPrecondition` ("encounter is not turn-based"). This gate lives in the handler, not in the toolkit verb (the toolkit verb assumes the caller has already mode-gated and produces an internal error if invariant violated; the handler converts to `FailedPrecondition`).
- Dispatch on `actionRef.GetId()`:
  - `"attack"`: extract `target.GetEntityId()`; validate it exists in `data.Monsters`; call `enc.TakeAction(core.PlayerID(playerID), actionRef, target)`. Toolkit errors: turn-violation → `FailedPrecondition`; unsupported-action → `Unimplemented`; target-not-found → `NotFound`.
  - default: `Unimplemented`.
- Save: `h.encRepo.Save(ctx, enc.ToData())` → `Internal` on error.
- Return `&encounterv2pb.TakeActionResponse{}` (empty — no `InputRequired` for plain attacks; events deliver outcomes).

### 2. `internal/handlers/dnd5e/v2/encounter/end_turn.go` (new file)

```go
func (h *Handler) EndTurn(ctx context.Context, req *encounterv2pb.EndTurnRequest) (*encounterv2pb.EndTurnResponse, error)
```

Required behavior:
- Auth + validation as above.
- Load encounter. Mode gate (`FailedPrecondition` if not `ModeTurnBased`).
- Call `enc.EndTurn(core.EntityID(req.GetEntityId()))`. Errors → `FailedPrecondition` (turn-violation, no-active-actor).
- **NPC dispatch loop** (orchestrator's responsibility):
  - After `EndTurn`, get the new active actor. If toolkit returns `(newActiveID, isNPC, err)` from `EndTurn`, use it. Otherwise call a query helper like `enc.ActiveActor()`.
  - If new active actor is an NPC (i.e., is in `data.Monsters`), call `enc.NPCAct(activeActor)`. Save. Then call `enc.EndTurn(activeActor)` to cycle past the NPC's turn (NPC always ends after `NPCAct` returns — this is a wave-2.8 simplification; richer turn ends are later).
  - Recurse: if the next active actor is also an NPC, repeat. Cap loop depth (e.g., 8 NPCs) to avoid runaway cycles. If the cap trips, log + return; the next manual EndTurn from any player picks up.
  - Once active actor is a player, return.
- Save final state.
- Return `&encounterv2pb.EndTurnResponse{}` (empty).

### 3. Translator extensions in `internal/handlers/dnd5e/v2/encounter/translate.go`

Add cases to `TranslateEvent`:

- `*events.DamageDealtEvent` → `EncounterEvent_EntityDamaged`:
  - `slice, ok := e.PerPlayer[viewer]; !ok || !slice.Visible` → `ErrViewerSawNothing`.
  - Map: `EntityId = string(e.TargetID)`, `Amount = int32(e.Amount)`, `DamageType = damageRefFor(e.DamageType)` (build a `*Ref` from the toolkit's damage-type string), `HpAfter = &HitPoints{Current: int32(e.HPAfter), Max: int32(e.MaxHP)}`, `SourceEntityId = ptr(string(e.SourceID))`.

- `*events.ConditionAppliedEvent` → `EncounterEvent_StatusApplied`:
  - Per-viewer slice gated as above.
  - Map to `StatusApplied{EntityId, Status: &StatusEffect{...from ConditionRef...}, SourceEntityId}`.

- `*events.AttackResolvedEvent` → for Wave 2.8, **drop** (return `ErrUnknownEventType` or a new `ErrSuppressed` typed error). The proto wire shape uses `EntityDamaged` as the canonical resolved-attack event; the cause-stage `AttackResolvedEvent` is for narration/animation hooks the web doesn't render in this wave. **Document the choice in PR description** so the close-the-loop can promote/refute the pattern.

  Alternative: emit a custom event type if a proto event exists for it. Verified at proto-pin v0.1.94: `events.proto` has no `AttackResolved` event. Drop it.

- `*events.ModeChangedEvent` → `EncounterEvent_ModeChanged`:
  - Audience is all players, so `Visible` is implicit. Map `From`/`To` from `core.EncounterMode` to proto `EncounterMode`. `Reason` passes through.

- `*events.TurnStartedEvent` → `EncounterEvent_TurnStarted`:
  - Map `EntityId = string(e.ActorID)`, `Round = int32(e.Round)`.

- `*events.TurnEndedEvent` → `EncounterEvent_TurnEnded`:
  - Map `EntityId = string(e.ActorID)`.

### 4. Tests

**Unit tests** — extend `internal/handlers/dnd5e/v2/encounter/handler_test.go`:
- `TestTakeAction_NoPlayerID_Unauthenticated`
- `TestTakeAction_MissingEncounter_NotFound`
- `TestTakeAction_NotTurnBased_FailedPrecondition`
- `TestTakeAction_AttackMissingTarget_NotFound`
- `TestTakeAction_NotYourTurn_FailedPrecondition`
- `TestTakeAction_UnsupportedAction_Unimplemented` (use any non-"attack" id)
- `TestTakeAction_AttackHappyPath` — saved encounter shows monster HP reduced.
- `TestEndTurn_NotTurnBased_FailedPrecondition`
- `TestEndTurn_NotYourTurn_FailedPrecondition`
- `TestEndTurn_HappyPath_CyclesToNextActor`
- `TestEndTurn_CyclesToNPC_DispatchesNPCAct` — uses a goblin fixture; confirms `NPCAct` is invoked and saved data shows monster moved + alice HP changed.

**Translator tests** — extend `internal/handlers/dnd5e/v2/encounter/translate_test.go`:
- `TestTranslateEvent_DamageDealt_HappyPath`
- `TestTranslateEvent_DamageDealt_NotVisible_ReturnsErrViewerSawNothing`
- `TestTranslateEvent_ConditionApplied_HappyPath`
- `TestTranslateEvent_ModeChanged_HappyPath`
- `TestTranslateEvent_TurnStarted_HappyPath`
- `TestTranslateEvent_TurnEnded_HappyPath`
- `TestTranslateEvent_AttackResolved_Suppressed` (verifies the drop choice; if you choose to keep emitting a custom proto field, adjust the test).

**Integration test** — extend `internal/integration/encounter_v2_test.go` per `pat-v2-bufconn-test-pattern`:

```go
func (s *EncounterV2IntegrationSuite) TestTakeAction_AndNPCTurn_TwoPlayersOneMonster() {
    enc := tkenc.New("enc-1", s.srv.BrokerV2)
    s.Require().NoError(enc.AddPlayer(tkenc.PlayerInput{
        PlayerID: "alice", EntityID: "char-alice", Position: core.Hex{Q: 0, R: 0, S: 0},
    }))
    s.Require().NoError(enc.AddPlayer(tkenc.PlayerInput{
        PlayerID: "bob", EntityID: "char-bob", Position: core.Hex{Q: 1, R: -1, S: 0},
    }))
    enc.AddMonster(tkenc.MonsterInput{
        ID: "goblin-1", Position: core.Hex{Q: 1, R: 0, S: -1},
        HP: 7, MaxHP: 7, AC: 15, Speed: 6, MonsterRef: "dnd5e:monsters:goblin",
        // DataJSON: serialized goblin with scimitar action
    })
    s.Require().NoError(enc.SetMode(core.ModeTurnBased))
    s.Require().NoError(s.srv.EncRepoV2.Save(s.ctx, enc.ToData()))

    ctxA := s.authCtx("alice")
    ctxB := s.authCtx("bob")

    streamA, err := s.srv.EncounterClientV2.StreamEncounter(ctxA, &encounterv2pb.StreamEncounterRequest{EncounterId: "enc-1"})
    s.Require().NoError(err)
    streamB, err := s.srv.EncounterClientV2.StreamEncounter(ctxB, &encounterv2pb.StreamEncounterRequest{EncounterId: "enc-1"})
    s.Require().NoError(err)

    // Drain snapshots.
    _, _ = streamA.Recv()
    _, _ = streamB.Recv()

    // alice attacks goblin
    _, err = s.srv.EncounterClientV2.TakeAction(ctxA, &encounterv2pb.TakeActionRequest{
        EncounterId: "enc-1",
        ActorEntityId: "char-alice",
        ActionRef: &commonpb.Ref{Module: "dnd5e", Type: "action", Id: "attack"},
        Target: &encounterv2pb.ActionTarget{Kind: &encounterv2pb.ActionTarget_EntityId{EntityId: "goblin-1"}},
    })
    s.Require().NoError(err)

    // Expect EntityDamaged on both streams (assuming visibility).
    // Use a drain helper to collect N events with timeout.
    // Assert: target=goblin-1, hp_after.current < 7 if hit.

    // alice ends turn — orchestrator should cycle to goblin, run NPCAct, then end goblin's turn
    _, err = s.srv.EncounterClientV2.EndTurn(ctxA, &encounterv2pb.EndTurnRequest{EncounterId: "enc-1", EntityId: "char-alice"})
    s.Require().NoError(err)

    // Expect on both streams: TurnEnded(alice), TurnStarted(goblin-1), EntityMoved(goblin-1), EntityDamaged(char-alice) (assuming hit), TurnEnded(goblin-1), TurnStarted(<next>).
}
```

Confirm visibility expectations against the toolkit's integration test fixtures. Adjust positions to guarantee both players see both attacks for the integration verify (per-viewer-projection edge cases get their own tests).

### 5. Toolkit pin bump

Update `go.mod`:
```
require github.com/KirkDiggler/rpg-toolkit/encounter v0.3.0
```

Plus any rulebook bump if rpg-toolkit#633 needed `dnd5e` touch. If not, leave at `v0.55.5`.

### 6. Documentation

Update `rpg-api/docs/architecture/components/encounter.md` (or equivalent) to list `TakeAction` + `EndTurn` under v2 RPCs, and document the NPC-dispatch-loop pattern.

## Acceptance

- `make ci-check` passes (no new lint regressions in `internal/handlers/dnd5e/v2/` or `internal/integration/`).
- All unit + translator + integration tests pass.
- The integration test exercises the full attack → endturn → npc-act → endturn loop on bufconn.

## Out of scope

- Mode flip RPC. Wave 2.8 uses `CreateEncounter(initial_mode=TURN_BASED)` (already in proto). If the playtest needs mid-encounter mode flip, file a followup; don't add a new RPC in this wave.
- Action types beyond "attack". Other `action_ref` ids return `Unimplemented`.
- `InputRequired` plumbing on `TakeAction`. Wave 2.9 (renumbered from 2.10).
- `SubmitCheck` RPC. Wave 2.9.
- Web consumer (filed separately).
- Reactions, multiattack, opportunity attacks.
- Death save state machine. `EntityDied` ships when HP hits 0 (toolkit-emitted); web rendering of post-death state is later.

## Anticipated patterns (close-the-loop will validate and append to role context)

- `pat-v2-mode-gating` — Handlers for `TakeAction` and `EndTurn` check `enc.Mode()` first and return `codes.FailedPrecondition` if not TURN_BASED.
- `pat-v2-npc-turn-dispatch` — After `EndTurn` advances initiative, the handler inspects the new active actor; if NPC, calls `enc.NPCAct(activeActor)` and recursively cycles through any subsequent NPC turns until the active actor is a player or initiative wraps.
- `pat-v2-combat-event-translator` — Mapping toolkit `DamageDealtEvent` → proto `EntityDamaged`, `ConditionAppliedEvent` → proto `StatusApplied`, `ModeChangedEvent`/`TurnStartedEvent`/`TurnEndedEvent` → corresponding proto events. Cause-stage events drop when the proto wire shape only carries effects.

## Cross-references

- Wave 2.8 tracker: rpg-project#23
- Wave plan: `rpg-project/ideas/encounter/v1alpha2/plans/06-wave-2.8-combat-slice.md`
- Toolkit pair: rpg-toolkit#633
- Web consumer pair: rpg-dnd5e-web#<filled at create>
- Wave 2.7 reference (cause/effect-event-split pattern): rpg-api#504
- Wave 2.6 reference (CreateEncounter / GetEncounter / snapshot): rpg-api#499, #500, #497
- Roadmap: `rpg-project/ideas/encounter/v1alpha2/roadmap.md`
- Design: `rpg-project/ideas/encounter/v1alpha2/design.md` §4 (TakeAction + EndTurn RPC shapes)
- Wave: 2.8 (Combat slice) on rpg-project board #11

### PR DESCRIPTION
**Pull request by KirkDiggler**
## Summary

The rpg-api half of Wave 2.8 (Combat slice). Pairs with the toolkit slice merged in `rpg-toolkit#638` (encounter v0.3.0). Closes #505.

- Bump `rpg-toolkit/encounter` to v0.3.0. Surfaces the toolkit's new combat verbs (`SetMode`, `TakeAction`, `EndTurn`, `NPCAct`), `MonsterData`, mode + initiative on `encounter.Data`, and the six new combat event types (`AttackResolvedEvent`, `DamageDealtEvent`, `ConditionAppliedEvent`, `ModeChangedEvent`, `TurnStartedEvent`, `TurnEndedEvent`).
- New v2 handlers `TakeAction` and `EndTurn` (`internal/handlers/dnd5e/v2/encounter/take_action.go`, `end_turn.go`). Both follow `pat-v2-handler-direct-no-orchestrator`: load → `LoadFromData` → call toolkit verb → save. Toolkit-emitted sentinels (`ErrNotTurnBased`, `ErrNotYourTurn`, `ErrUnsupportedAction`, `ErrUnknownTarget`, `ErrNonCombatant`, `ErrNoCombatants`) are mapped per `pat-v2-status-code-mapping`.
- `EndTurn` drives the orchestrator-side **NPC dispatch loop** (`pat-v2-npc-turn-dispatch`). The toolkit's `EndTurn` returns `(newActive, isNPC, err)`; when the new active is an NPC the handler runs `NPCAct` + a follow-up `EndTurn` server-side, looping until the active actor is a player. Loop is capped at 16 to defend against pathological all-NPC initiatives.
- Translator additions in `translate.go` for the six new event types. `DamageDealt → EntityDamaged`, `ConditionApplied → StatusApplied`, `ModeChanged/TurnStarted/TurnEnded` → matching proto envelopes. `AttackResolvedEvent` has no proto wire shape (proto designers chose `EntityDamaged` as the canonical effect-event), so the translator suppresses it via a new `ErrEventSuppressed` and the stream loop drops without log.
- `CreateEncounter` now accepts `INITIAL_MODE_TURN_BASED`, delegating initiative roll to the toolkit's `SetMode`.
- Bufconn integration test (`TestCombatSlice_TakeActionAndEndTurn_NPCDispatch`) exercises the full attack → endturn → npc-act → endturn cycle with two players + one monster. Uses a per-stream `eventSink` with snapshot watermarks so live events are deterministically separated from initial-state replay.

## Behavior

- `TakeAction(playerID, actor_entity_id, action_ref={dnd5e:action:attack}, target.entity_id=monster)` — gated to `TURN_BASED`. Returns `FailedPrecondition` for state-dependent failures (mode, turn, non-combatant, unknown-target); `Unimplemented` for unsupported action ids; `PermissionDenied` for actor mismatch / non-member; `InvalidArgument` for malformed request.
- `EndTurn(playerID, entity_id)` — gated to `TURN_BASED`. After advancing initiative, runs the NPC dispatch loop until next active actor is a player.
- Visibility per `pat-v2-translator-typed-errors`: per-viewer `Visible:false` returns `ErrViewerSawNothing` (silent skip); `AttackResolvedEvent` returns `ErrEventSuppressed` (silent suppress).

## Deferred reviewer notes

- The toolkit's stand-in attack resolver (run by `NPCAct` and `TakeAction` for player attacks) is captured at `rpg-toolkit#635` — the dnd5e action layer doesn't yet route through `combat.ResolveAttack`. Not blocking; the wire shape is correct. When `#635` lands, only the toolkit changes; this PR's translator stays as-is.
- `TestCreateEncounter_TurnBasedMode_RollsInitiative` replaces the old `TestCreateEncounter_TurnBasedMode_InvalidArgument` (TURN_BASED is now accepted on create).

## Test plan

- [x] `go test ./internal/handlers/dnd5e/v2/encounter/...` — unit tests pass.
- [x] `go test ./internal/integration/... -run TestEncounterV2IntegrationSuite` — bufconn integration including the new combat slice test passes.
- [x] `golangci-lint run ./internal/handlers/dnd5e/v2/encounter/...` — 0 issues in new code (repo has 66 pre-existing lint issues unrelated to this PR; same count on `main`).
- [x] `go mod tidy` clean; `gofmt -s -l` clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/handlers/dnd5e/v2/encounter/end_turn.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+	// NPC dispatch loop. After the toolkit's EndTurn advances initiative, if
+	// the new active actor is an NPC the handler runs its turn server-side
+	// and ends it, then re-checks. The cap protects against the pathological
+	// all-NPC initiative case.
+	for depth := 0; isNPC && depth < maxNPCChainDepth; depth++ {
+		if actErr := enc.NPCAct(ctx, newActive); actErr != nil {
+			// NPCAct errors are surfaced as Internal — they indicate either a
+			// rehydration / bus / publish failure (system-shaped) or an
+			// unexpected state mismatch. Save what state we have first so the
```
[the remark this answers is not in the document]
Comment (KirkDiggler) [reply internal/handlers/dnd5e/v2/encounter/end_turn.go]: Fixed in ec14dfb. Cap is now `len(data.Initiative) + npcChainSafetyMargin` so all-NPC rosters cycle fully rather than freezing. Added a defensive `FailedPrecondition` if the cap is hit while `isNPC` is still true — surfaces misconfigured (zero-player) encounters loudly instead of silently.

File: internal/handlers/dnd5e/v2/encounter/translate.go [hunk 2]
```diff
@@ -166,6 +193,198 @@ func translateEntityDisappearedEvent(e *events.EntityDisappearedEvent, viewer co
...
+	for i := 0; i < len(s); i++ {
+		if s[i] == ':' {
+			parts = append(parts, s[start:i])
+			start = i + 1
+			colons++
+		}
+	}
+	if colons != 2 {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (KirkDiggler) [reply internal/handlers/dnd5e/v2/encounter/translate.go:405]: Fixed in ec14dfb — aligned the doc comment to match the nil return.

File: internal/integration/encounter_v2_test.go [hunk 3]
```diff
@@ -524,6 +525,253 @@ func (s *EncounterV2IntegrationSuite) TestStreamEncounter_LiveEventsDoNotDuplica
...
 }
 
+// TestCombatSlice_TakeActionAndEndTurn_NPCDispatch is the wave-2.8 gate test.
+// It exercises the full attack → endturn → npc-act → endturn loop on bufconn:
+//
+//   - Two players + one monster, all in mutual LoS, mode set to TURN_BASED on
+//     seed (initiative is rolled by the toolkit).
+//   - Both players subscribe via StreamEncounter and drain replay events.
+//   - The active player attacks goblin-1; both streams must observe an
+//     EntityDamaged event (per-viewer projection routes them based on LoS).
```
[the remark this answers is not in the document]
Comment (KirkDiggler) [reply internal/integration/encounter_v2_test.go]: Updated the doc-comment in ec14dfb to clarify that EntityDamaged arrives on hit only (real d20 resolution → miss-runs are valid passes). Keeping the test non-deterministic for now is intentional: the toolkit's stand-in resolver will be replaced when rpg-toolkit#635 lands, and forcing a hit here would require fixture surgery that doesn't survive that migration.

File: internal/handlers/dnd5e/v2/encounter/combat_handlers_test.go [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+		ID: monsterGoblin1, Position: core.Hex{Q: 2, R: -1, S: -1},
+		HP: 7, MaxHP: 7, AC: 15, Speed: 6,
+		MonsterRef:  "dnd5e:monsters:goblin",
+		AttackBonus: 4, DamageDice: "1d6+2", DamageType: "slashing",
+	}))
+	s.Require().NoError(enc.SetMode(core.ModeTurnBased))
+	s.Require().NoError(s.repo.Save(s.ctx, enc.ToData()))
+}
+
+// makeAttackTargetMonster constructs the standard "alice attacks goblin-1"
```
[the remark this answers is not in the document]
Comment (KirkDiggler) [reply internal/handlers/dnd5e/v2/encounter/combat_handlers_test.go]: Fixed in ec14dfb — updated the comment to match the actual function name.

File: internal/integration/encounter_v2_test.go [hunk 5]
```diff
@@ -524,6 +525,253 @@ func (s *EncounterV2IntegrationSuite) TestStreamEncounter_LiveEventsDoNotDuplica
...
+	// active RPC. This avoids racy multi-call collectStreamEvents patterns.
+	aSink := newEventSink()
+	bSink := newEventSink()
+	go aSink.drain(streamA)
+	go bSink.drain(streamB)
+
+	// Wait briefly for replay to land on both sinks.
+	time.Sleep(300 * time.Millisecond)
+	aSink.snapshot() // capture replay watermark; subsequent reads start AFTER this
+	bSink.snapshot()
```
[the remark this answers is not in the document]
Comment (KirkDiggler) [reply internal/integration/encounter_v2_test.go]: Fixed in ec14dfb. Replaced the fixed sleep with `Eventually` polling for replay completion (SnapshotDelivered + EntityAppeared + GeometryRevealed all present). Watermark is now set deterministically without timing assumptions.

**Comment by KirkDiggler** [comment]
The failing \`Build and Push Docker Image\` CI run is a flake in v1alpha1 \`TestMonkIntegrationSuite/TestMartialArts_CanAttackInCombat\`, unrelated to this PR. Filed as #508.

**Diagnostic summary:**
- Companion \`Test and Coverage\` workflow passed on the same commit
- Cannot reproduce locally (5/5 isolated runs + 3/3 full v1 encounter suite runs all pass)
- \`go.sum\` diff vs main shows only the \`encounter v0.2.0 → v0.3.0\` bump — no transitive changes to \`rulebooks/dnd5e\`, \`tools/environments\`, or \`tools/spatial\`
- The encounter v0.3.0 module is only imported by the v2 path; the failing v1alpha1 monk test routes through the v1 orchestrator + dungeon component which don't import it
- Hypothesis: \`StartCombat\` uses random dungeon generation; when the seed produces a monster-less starting room, \`FindMonsterTarget\` (helpers.go:316) fatals with the observed message

Re-ran the failed build job (25579454076) to confirm green path.
